atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Cell.java

```diff
@@ -62,10 +67,18 @@ public static boolean isNameValid(byte[] name) {
         return name != null && name.length > 0 && name.length <= MAX_NAME_LENGTH;
     }
 
-    public static void validateNameValid(byte[] name) {
+    private void validateNameValid(byte[] name) {
         Preconditions.checkNotNull(name, "name cannot be null");
         Preconditions.checkArgument(name.length > 0, "name must be non-empty");
-        Preconditions.checkArgument(name.length <= MAX_NAME_LENGTH, "name must be no larger than " + MAX_NAME_LENGTH);
+
+        String lengthErrorMessage = "name must be no longer than " + MAX_NAME_LENGTH;
+        if (log.isDebugEnabled()) {
+            lengthErrorMessage += ". Cell creation that was attempted was: " + this
+                    + "; since the vast majority of people encountering this problem are using unbounded Strings as"
+                    + " components, it may aid your debugging to know the ASCII interpretation of the bad field was:"
+                    + " [" + new String(name, StandardCharsets.US_ASCII) + "]";
```

Comment:
From an internal logging collection perspective, we require products to use a findbugs plugin (http://kengotoda.github.io/findbugs-slf4j/) to ensure secure logging syntax. In this case, the logged string should use placeholders instead of object concatenation. For example:

log.warn("Cell creation that was attempted was: {}... etc", this)